#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_ftbfs_amd64.dpatch by LI Daobing <lidaobing@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix FTBFS under AMD64
## DP: See: http://code.google.com/p/iptux/issues/detail?id=5

@DPATCH@

--- iptux-0.3.3.orig/src/utils.cpp
+++ iptux-0.3.3/src/utils.cpp
@@ -42,7 +42,7 @@
 	size_t size, len;
 
 	cd = iconv_open(tocode, fromcode);
-	if ((int)cd == -1) {
+	if (cd == iconv_t(-1)) {
 		pwarning(Fail,
 			 _("act: create encode conversion table,warning: %s\n"),
 			 strerror(errno));
